testing.py

```diff
@@ -325,6 +325,12 @@ def queue_response(self, response_code, media_type="text/html",
     def do_get(self, url, headers=None, allowed_response_codes=None, **kwargs):
         self.requests.append(url)
         response = self.responses.pop()
+        if isinstance(response.status_code, Exception):
+            raise response.status_code
```

Comment:
This lets you simulate an exception (such as a timeout) that happened during the request.